comp/core/agenttelemetry/impl/agenttelemetry.go

```diff
@@ -276,44 +276,52 @@ func buildKeysForMetricsPreviousValues(mt dto.MetricType, metricName string, met
 			keyName = builder.String()
 		}
 
+		// Add bucket names to the key
 		if mt == dto.MetricType_HISTOGRAM {
 			// add bucket names to the key
 			for _, bucket := range m.Histogram.GetBucket() {
 				keyNames = append(keyNames, fmt.Sprintf("%v:%v", keyName, bucket.GetUpperBound()))
 			}
-		} else {
-			keyNames = append(keyNames, keyName)
 		}
+
+		// For regular metric (and for HISTOGRAM +Inf bucket which follows the last bucket)
+		keyNames = append(keyNames, keyName)
```

Comment:
Should this be inside the for loop that starts at line 260? If `metrics` has 2 items (for example) and neither has tags, then `keyName` gets set to the `metricName` both iterations of the loop, and then appended to `keyNames` twice.

Reply:
Indeed @dustmop that would be a problem. And visually it looks like that, but if you look carefully the line is inside the loop. Still, thank you for noticing that, I fall to the same conclusion by looking on the diff here.

At any rate, I have used this as an opportunity to improve code a bit more and write a few more unit tests. Thank you again.